src/encoded/types/shared_calculated_properties.py

```diff
@@ -536,31 +552,15 @@ def replication_type(self, request, replicates=None, assay_term_name=None):
                     # I cannot make a call about replicate structure
                     return None
             else:
-                # REPLICATES WITH NO LIBRARIES WILL BE CAUGHT BY AUDIT (TICKET 3268)
                 # If I have a replicate without a library,
                 # I cannot make a call about the replicate structure
                 return None
 
-        #  exclude ENCODE2
-        if (len(set(biosample_number_list)) < 2):
+        if (len(biosample_numbers_set) < 2):
             return 'unreplicated'
 
-        if biosample_type == 'cell line':
-            return 'isogenic'
-
-        # Since we are not looking for model organisms here, we likely need audits
-        if biosample_species != '/organisms/human/':
-            if len(set(biosample_donor_list)) == 1:
-                return 'isogenic'
-            else:
-                return 'anisogenic'
-
-        if len(set(biosample_donor_list)) == 0:
+        if len(biosample_donors_set) == 0:
             return None
-        if len(set(biosample_donor_list)) == 1:
-            if None in biosample_donor_list:
-                return None
-            else:
-                return 'isogenic'
-
+        if len(biosample_donors_set) == 1:
+            return 'isogenic'
```

Comment:
I tried to clean it up a little bit. I'm wondering if the bind-n-seq stuff needs to be where it is? Do we only call bind-n-seq unreplicated if one of its replicates doesn't have a biosample? Or is this always true which means we can return ureplicated near the top if it is a bind-n-seq?

```python
def replication_type(self, request, original_files=None, replicates=None, assay_term_name=None):
        if not original_files:
            return None
        if not replicates:
            return None
        excluded_statuses = (
            'uploading',
            'content error',
            'upload failed',
            'deleted',
            'replaced',
        )
        filtered_file_objects = {
            request.embed(f, '@@object')
            for f in paths_filtered_by_status(
                    request,
                    original_files,
                    exclude=excluded_statuses
            )
        }
        related_replicates_from_filtered_files = {
            f.get('replicate')
            for f in filtered_file_objects
            if f.get('replicate') and f.get('replicate') in replicates
        }
        filtered_replicate_objects = {
            request.embed(r, '@@object')
            for r in paths_filtered_by_status(
                    request,
                    related_replicates_from_filtered_files,
                    exclude=excluded_statuses
            )
        }
        bio_rep_dict = {}
        for r in filtered_replicate_objects:
            bio_rep_num = r['biological_replicate_number']
            if bio_rep_num not in bio_rep_dict:
                bio_rep_dict[bio_rep_num] = r
                continue
            tech_rep_num = r['technical_replicate_number']
            if tech_rep_num < bio_rep_dict[bio_rep_num]['technical_replicate_number']:
                bio_rep_dict[bio_rep_num] = r
        biosample_donors_set = set()
        biosample_numbers_set = set()
        for replicate_object in bio_rep_dict.values():
            libraries = r.get('libraries')
            if not libraries:
                # If I have a replicate without a library I cannot make a
                # call about the replicate structure.
                return None
            biosamples = request.select_distinct_values(
                'biosample', *libraries
            )
            if not biosamples:
                # Special treatment for "RNA Bind-n-Seq" they will be called
                # unreplicated until we change our mind.
                if assay_term_name == 'RNA Bind-n-Seq':
                    return 'unreplicated'
                # If I have a library without a biosample I cannot make a
                # call about replicate structure.
                return None
            for b in biosamples:
                biosample_object = request.embed(b, '@@object')
                biosample_donors_set.add(biosample_object.get('donor'))
                biosample_numbers_set.add(replicate_object.get('biological_replicate_number'))
        if len(biosample_numbers_set) < 2:
            return 'unreplicated'
        if len(biosample_donors_set) == 0:
            return None
        if len(biosample_donors_set) == 1:
            return 'isogenic'
        return 'anisogenic'
```

Reply:
Also is the step to get the lowest tech_rep number necessary to count libraries and biosamples later?  (Apparently it is according to the comment I wasn't sure was relevant anymore.)

Reply:
yeah, the comment is relevant, and that is the behavior supporting future removal of technical replicates

Reply:
I have the same question about RNA BNS.

Reply:
If we want to take RBNS out - we will need to add special condition that makes sure there are libraries in these experiments. Otherwise, I think it happens once we know the experiment has libraries.